pkg/stream/xprotocol/connpool_multiplex.go

```diff
@@ -136,14 +159,17 @@ func (p *poolMultiplex) NewStream(ctx context.Context, receiver types.StreamRece
 
 // Shutdown stop the keepalive, so the connection will be idle after requests finished
 func (p *poolMultiplex) Shutdown() {
-	f := func(k, v interface{}) bool {
-		ac, _ := v.(*activeClientMultiplex)
-		if ac.keepAlive != nil {
-			ac.keepAlive.keepAlive.Stop()
+	atomic.StoreInt64(&p.shutdown, 1)
+	for i := 0; i < len(p.activeClients); i++ {
```

Comment:
for _, client := range p.activeclient

Reply:
![image](https://user-images.githubusercontent.com/384546/93760618-f82b9880-fc3e-11ea-915a-b04d3b4e2aba.png)

This grammar will cause the lock in sync.Map to be copied